### ISSUE TITLE
Two Factor Authentication enabled. User is brought to User Management page after login

### ISSUE THREAD

**Issue by afatac** [salesagility/SuiteCRM#5836]
#### Issue

See the post in the forum
https://suitecrm.com/suitecrm/forum/suitecrm-7-0-discussion/18965-two-factor-authentication-enabled-user-is-brought-to-user-management-page-after-login

#### Expected Behavior

Dashboard page should be shown

#### Actual Behavior

User Management page is shown instead

This message is shown in suitecrm.log
Tue May  8 09:40:27 2018 [1990118][1][FATAL] DEBUG: token already sent

#### Steps to Reproduce

1.
2.
3.
4.

#### Your Environment

* SuiteCRM Version used: 7.10.4
* Browser name and version (e.g. Chrome Version 51.0.2704.63 (64-bit)): Opera 52.0.2871.99 (64-bit)
* Environment name and version (e.g. MySQL, PHP 7): MySQL, PHP 7
* Operating System and version (e.g Ubuntu 16.04): Shared Linux host

**Comment by Ramblin** [comment]
Has this issue been resolved?

I have just loaded v7.10.11 on a fresh Debian 9 install and I am having the same issue.  Using Firefox or Chrome does not make a difference.

When I activated two-factor authentication, users were thereafter directed to their profile page (not the Edit view, just the normal view) upon login.

They can still go to the home page but it takes an extra click and I'd rather they started on their Home Page

Now here's the twist: I disabled two-factor authentication for the user (as a test) and they still go to the Profile page on login. They did NOT do that before.

If this issue has not been resolved, is there a way - until this gets resolved, that I can use some kind of logic_hook to force the system to take a user to the Home page after login?  (if so, please be specific; I am not all that familiar with the coding of SuiteCRM but I am familiar with php, ... so if you tell me what to do I can do it)

**Comment by Mister-A** [comment]
Issue remains a year later on 7.11.10

We should all be using 2FA these days so I can't agree this is a low priority.

Any chance of a bump?

**Comment by samus-aran** [comment]
@Mister-A @Ramblin That is a fix proposed. Are you able to have a test to confirm this works for yous?

**Comment by Mister-A** [comment]
@samus-aran Confirmed working for me. 2FA login now correctly redirects to home index (dashboard) instead of user index. Many thanks @mattlorimer

**Comment by Ramblin** [comment]
Worked like a charm - thank you very much!
2FA is now something I will use by default for my users.

I did this on one installation at v7.11.12 and one on v7.10.22 and both operated correctly after the patch, with the User going to the Home page on login with 2FA.

I see two points of edit to the file, but the first point of edit actaully has no changes from the red-shaded (deleted) section and the green-shaded (added) section, so all I did was change
            <form method="post" action="index.php?module=Users&action=index">
to
            <form method="post">
and all is now good.

Thanks again

**Comment by Ramblin** [comment]
Ashley,

 

Worked like a charm - thank you very much!

2FA is now something I will use by default for my users.

 

I did this on one installation at v7.11.12 and one on v7.10.22 and both operated correctly after the patch, with the User going to the Home page on login with 2FA.

 

I see two points of edit to the file, but the first point of edit actaully has no changes from the red-shaded (deleted) section and the green-shaded (added) section, so all I did was change

<form method="post" action="index.php?module=Users&action=index">

to

<form method="post">

and all is now good.

 

Thanks again

 

Richard Cantin

 <mailto:richard.cantin@ayuda.ca> richard.cantin@ayuda.ca

(519) 957-2414 [office]

(519) 574-9401 [mobile]
